### PR TITLE
Apps and Config enhancements

### PR DESCRIPTION
**Pull request by weirdaze**
added the capability to rename and clone apps
added the ability to add env vars to the global app and to unset and toggle restart when setting env vars for all apps. if a key is both in the config dict AND in the unset list, then env var will first be added and then deleted. all of these functions will respect the restart instruction.
tried to make error handling as descriptive as possible so that errors can carry through from the various functions instead of from a single function. Have used it several times now and it works well with dokku v 0.19.x

### CODE REVIEW

File: library/dokku_app.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
     default: present
     choices: [ "present", "absent" ]
     aliases: []
+modified by: Elliott Castillo
```
Comment (josegonzalez) [inline library/dokku_app.py]: Please remove this, that is what git blame is for

File: library/dokku_config.py [hunk 2]
```diff
@@ -69,79 +85,159 @@ def subprocess_check_output(command, split='\n'):
...
+    # error = output
     return output, error
 
 
 def dokku_config_set(data):
     is_error = True
     has_changed = False
     meta = {'present': False}
-
+    meta = {'error1': None}
```
Comment (josegonzalez) [inline library/dokku_config.py]: Once you move `unset` to another command, this should go back to `error`

File: library/dokku_config.py [hunk 3]
```diff
@@ -69,79 +85,159 @@ def subprocess_check_output(command, split='\n'):
...
+    options = []
```
Comment (josegonzalez) [inline library/dokku_config.py]: Options isn't used till later, so we can move this down.

File: library/dokku_config.py [hunk 4]
```diff
@@ -69,79 +85,159 @@ def subprocess_check_output(command, split='\n'):
...
+    
+    if not data['restart']:
+        options.append('--no-restart') 
+    
+    # you can further modify this to add other params such as "encoded" or other options in future dokku versions
+
+    if data['app'] == '--global':
+        command = 'dokku config:set {1} --global {0}'.format(' '.join(values), ' '.join(options))
+    else:
+        command = 'dokku config:set {2} {0} {1}'.format(data['app'], ' '.join(values), ' '.join(options))
```
Comment (josegonzalez) [inline library/dokku_config.py]: This whole block can just be:

```
command = 'dokku config:set {0} {1} {2}'.format(' '.join(options), data['app'], ' '.join(values))
```

File: library/dokku_app.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (weirdaze) [reply library/dokku_app.py]: Ok. Sorry, this is literally my first pull request ever.

[the remark this answers is not in the document]
Comment (josegonzalez) [reply library/dokku_app.py]: No worries! It's a great feature and I'm looking forward to having it in our software :)

File: library/dokku_config.py (hunk 4 above)
Comment (weirdaze) [reply library/dokku_config.py]: Yeah, good point. That originally applied until I realized the app could just be named “—global” and then I forgot to change it.

**Comment by josegonzalez** [comment]
Closing as there hasn't been any follow-up and there are now merge conflicts. Feel free to resubmit while incorporating feedback.

**Pull request closed without merging**